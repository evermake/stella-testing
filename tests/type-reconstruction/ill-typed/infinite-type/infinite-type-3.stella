language core;

extend with #type-reconstruction;
  
fn main(n : auto) -> auto {
  return n(n);
}
