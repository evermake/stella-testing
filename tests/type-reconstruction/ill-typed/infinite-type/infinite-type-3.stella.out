An error occurred during typechecking!
Type Error Tag: [ERROR_OCCURS_CHECK_INFINITE_TYPE]
cannot construct an infinite type when type
  ?T3
is expected to unify with
  fn (?T3) -> ?T4
when typechecking the expression
  n